lib/services/cloud-build-service.ts

```diff
@@ -471,7 +476,7 @@ export class CloudBuildService extends EventEmitter implements ICloudBuildServic
 		if (!runtimeVersion && coreModulesVersion) {
 			// no runtime added. Let's find out which one we need based on the tns-core-modules.
 			if (semver.valid(coreModulesVersion)) {
-				runtimeVersion = `${semver.major(coreModulesVersion)}.${semver.minor(coreModulesVersion)}.*`;
+				runtimeVersion =  await this.getLatestMatchingVersion(runtimePackageName, `~${semver.major(coreModulesVersion)}.${semver.minor(coreModulesVersion)}.0`);
```

Comment:
Not a merge-stopper but you could extract a helper method `versionString => ~${semver.major(versionString)}.${semver.minor(versionString)}.0`